[BUG] panic: ast.Walk when I try to instrument.

<!--
If you have identified a specific bug in the orchestrion codebase, please describe it here. If the
issue requires debugging or investigation of your specific private setup, that is best accomplished via support. Please contact Datadog [support](http://docs.datadoghq.com/help/).
-->

**Version of orchestrion**

**v0.4.1**
<!-- If you aren't using the [latest version](https://github.com/DataDog/orchestrion/releases) of orchestrion, try upgrading first to see if your issue has already been resolved -->

**Describe what happened:**
I would like to instrument a project that I regularly use. But I've a panic. 
```
panic: ast.Walk: unexpected node type <nil>

goroutine 1 [running]:
github.com/dave/dst.Walk({0x727d38?, 0xc0005a7800?}, {0x0?, 0x0?})
        /home/bagl3y/go/pkg/mod/github.com/dave/dst@v0.27.2/walk.go:332 +0x1894
github.com/dave/dst.walkStmtList(...)
        /home/bagl3y/go/pkg/mod/github.com/dave/dst@v0.27.2/walk.go:32
github.com/dave/dst.Walk({0x727d38?, 0xc0005a7800?}, {0x727ed8?, 0xc00017be00?})
        /home/bagl3y/go/pkg/mod/github.com/dave/dst@v0.27.2/walk.go:216 +0x1ca5
github.com/dave/dst.Walk({0x727d38?, 0xc0005a7800?}, {0x728398?, 0xc0001103c0?})
        /home/bagl3y/go/pkg/mod/github.com/dave/dst@v0.27.2/walk.go:315 +0x1725
github.com/dave/dst.walkDeclList(...)
        /home/bagl3y/go/pkg/mod/github.com/dave/dst@v0.27.2/walk.go:38
github.com/dave/dst.Walk({0x727d38?, 0xc0005a7800?}, {0x7283b8?, 0xc0000f0410?})
        /home/bagl3y/go/pkg/mod/github.com/dave/dst@v0.27.2/walk.go:321 +0x18e5
github.com/dave/dst.Inspect(...)
        /home/bagl3y/go/pkg/mod/github.com/dave/dst@v0.27.2/walk.go:353
github.com/dave/dst/decorator.(*FileRestorer).updateImports(0xc00045c2d0)
        /home/bagl3y/go/pkg/mod/github.com/dave/dst@v0.27.2/decorator/restorer.go:190 +0x176
github.com/dave/dst/decorator.(*FileRestorer).RestoreFile(0xc00045c2d0, 0x0?)
        /home/bagl3y/go/pkg/mod/github.com/dave/dst@v0.27.2/decorator/restorer.go:135 +0x23e
github.com/dave/dst/decorator.(*Restorer).RestoreFile(...)
        /home/bagl3y/go/pkg/mod/github.com/dave/dst@v0.27.2/decorator/restorer.go:65
github.com/dave/dst/decorator.(*Restorer).Fprint(0xc0005a8480, {0x727c58, 0xc000003890}, 0x6c18c0?)
        /home/bagl3y/go/pkg/mod/github.com/dave/dst@v0.27.2/decorator/restorer.go:56 +0x93
github.com/datadog/orchestrion/internal/instrument.InstrumentFile({0xc000164828, 0x16}, {0x727ab8?, 0xc000324788}, {{0x6c8f9b?, 0xc000164828?}})
        /home/bagl3y/go/pkg/mod/github.com/datadog/orchestrion@v0.4.1/internal/instrument/instrument.go:103 +0x777
github.com/datadog/orchestrion/internal/instrument.ProcessPackage.func1({0xc000164828, 0x16}, {0x72af90?, 0xc0001b3440?}, {0x0?, 0x0?})
        /home/bagl3y/go/pkg/mod/github.com/datadog/orchestrion@v0.4.1/internal/instrument/instrument.go:48 +0x20c
io/fs.walkDir({0x727bb8, 0xc0000421a0}, {0xc000164828, 0x16}, {0x72af90, 0xc0001b3440}, 0xc0004fbdc8)
        /home/linuxbrew/.linuxbrew/Cellar/go/1.21.3/libexec/src/io/fs/walk.go:73 +0x6c
io/fs.walkDir({0x727bb8, 0xc0000421a0}, {0xc000128d10, 0xf}, {0x72af90, 0xc0001b3180}, 0xc0004fbdc8)
        /home/linuxbrew/.linuxbrew/Cellar/go/1.21.3/libexec/src/io/fs/walk.go:95 +0x2c5
io/fs.walkDir({0x727bb8, 0xc0000421a0}, {0xc000128ca0, 0xb}, {0x72af90, 0xc0001bd0c0}, 0xc0004fbdc8)
        /home/linuxbrew/.linuxbrew/Cellar/go/1.21.3/libexec/src/io/fs/walk.go:95 +0x2c5
io/fs.walkDir({0x727bb8, 0xc0000421a0}, {0xc00026fb10, 0x3}, {0x72af90, 0xc0000c66c0}, 0xc0004fbdc8)
        /home/linuxbrew/.linuxbrew/Cellar/go/1.21.3/libexec/src/io/fs/walk.go:95 +0x2c5
io/fs.walkDir({0x727bb8, 0xc0000421a0}, {0x726af4, 0x1}, {0x72afc8, 0xc0000421b0}, 0xc0004fbdc8)
        /home/linuxbrew/.linuxbrew/Cellar/go/1.21.3/libexec/src/io/fs/walk.go:95 +0x2c5
io/fs.WalkDir({0x727bb8, 0xc0000421a0}, {0x726af4, 0x1}, 0xc000185dc8)
        /home/linuxbrew/.linuxbrew/Cellar/go/1.21.3/libexec/src/io/fs/walk.go:122 +0xd2
github.com/datadog/orchestrion/internal/instrument.ProcessPackage({0xc0000305c0, 0x32}, 0x6dd7d8, 0x6dd7f8, {{0x6c8f9b?, 0x1?}})
        /home/bagl3y/go/pkg/mod/github.com/datadog/orchestrion@v0.4.1/internal/instrument/instrument.go:36 +0xd6
main.main()
        /home/bagl3y/go/pkg/mod/github.com/datadog/orchestrion@v0.4.1/main.go:77 +0x4e9
```
**Describe what you expected:**
Instrument my project

**Steps to reproduce the issue:**
Clone https://github.com/gotenberg/gotenberg and try to instrument it (`orchestrion -w ./`)
<!-- If you can include a minimal code sample to reproduce the issue it can significantly reduce resolution time -->

**Additional environment details (Version of Go, Operating System, etc.):**

Golang 1.21.3 linux/amd64 (Same issue in a Docker Build)

--- a/internal/instrument/chiv5.go
+++ b/internal/instrument/chiv5.go
@@ -10,30 +10,13 @@ import (
 )
 
 func instrumentChiV5(stmt *dst.AssignStmt) []dst.Stmt {
-	if !isChiV5(stmt) {
-		return nil
-	}
-	stmt.Decorations().Start.Prepend(dd_instrumented)
-	return []dst.Stmt{
-		stmt,
-		chiV5Middleware(stmt),
-	}
+	return instrumentMiddleware(stmt, isChiV5, "ChiV5Middleware")
 }
 
 func isChiV5(stmt *dst.AssignStmt) bool {
 	rhs := stmt.Rhs[0]
 	f, ok := funcIdent(rhs)
 	return ok && f.Path == "github.com/go-chi/chi/v5" && f.Name == "NewRouter"
-}
-
-func chiV5Middleware(got *dst.AssignStmt) dst.Stmt {
-	iden, ok := got.Lhs[0].(*dst.Ident)
-	if !ok {
-		return nil
-	}
-	stmt := useMiddleware(iden.Name, "ChiV5Middleware")
-	markAsInstrumented(stmt)
-	return stmt
 }
 
 // removeChiV5 returns whether a statement corresponds to orchestrion's Chi-middleware registration

--- a/internal/instrument/chiv5_test.go
+++ b/internal/instrument/chiv5_test.go
@@ -50,6 +50,7 @@ func register() {
 		{pkg: `"github.com/go-chi/chi/v5"`, stmt: `r := chi.NewRouter()`, want: `r.Use(instrument.ChiV5Middleware())`, tmpl: wantTpl},
 		{pkg: `chi "github.com/go-chi/chi/v5"`, stmt: `r := chi.NewRouter()`, want: `r.Use(instrument.ChiV5Middleware())`, tmpl: wantTpl},
 		{pkg: `chiv5 "github.com/go-chi/chi/v5"`, stmt: `r := chiv5.NewRouter()`, want: `r.Use(instrument.ChiV5Middleware())`, tmpl: wantTpl},
+		{pkg: `"github.com/go-chi/chi/v5"`, stmt: `api.server = chi.NewRouter()`, want: `api.server.Use(instrument.ChiV5Middleware())`, tmpl: wantTpl},
 	}
 
 	for i, tc := range tests {

--- a/internal/instrument/echov4.go
+++ b/internal/instrument/echov4.go
@@ -8,30 +8,13 @@ package instrument
 import "github.com/dave/dst"
 
 func instrumentEchoV4(stmt *dst.AssignStmt) []dst.Stmt {
-	if !isEchoV4(stmt) {
-		return nil
-	}
-	stmt.Decorations().Start.Prepend(dd_instrumented)
-	return []dst.Stmt{
-		stmt,
-		echoV4Middleware(stmt),
-	}
+	return instrumentMiddleware(stmt, isEchoV4, "EchoV4Middleware")
 }
 
 func isEchoV4(stmt *dst.AssignStmt) bool {
 	rhs := stmt.Rhs[0]
 	f, ok := funcIdent(rhs)
 	return ok && f.Path == "github.com/labstack/echo/v4" && f.Name == "New"
-}
-
-func echoV4Middleware(got *dst.AssignStmt) dst.Stmt {
-	iden, ok := got.Lhs[0].(*dst.Ident)
-	if !ok {
-		return nil
-	}
-	stmt := useMiddleware(iden.Name, "EchoV4Middleware")
-	markAsInstrumented(stmt)
-	return stmt
 }
 
 // removeEchoV4 returns whether a statement corresponds to orchestrion's Echo-middleware registration

--- a/internal/instrument/echov4_test.go
+++ b/internal/instrument/echov4_test.go
@@ -50,6 +50,7 @@ func register() {
 		{pkg: `"github.com/labstack/echo/v4"`, stmt: `r := echo.New()`, want: `r.Use(instrument.EchoV4Middleware())`, tmpl: wantTpl},
 		{pkg: `echo "github.com/labstack/echo/v4"`, stmt: `r := echo.New()`, want: `r.Use(instrument.EchoV4Middleware())`, tmpl: wantTpl},
 		{pkg: `echov4 "github.com/labstack/echo/v4"`, stmt: `r := echov4.New()`, want: `r.Use(instrument.EchoV4Middleware())`, tmpl: wantTpl},
+		{pkg: `"github.com/labstack/echo/v4"`, stmt: `api.server = echo.New()`, want: `api.server.Use(instrument.EchoV4Middleware())`, tmpl: wantTpl},
 	}
 
 	for i, tc := range tests {

--- a/internal/instrument/gin.go
+++ b/internal/instrument/gin.go
@@ -8,30 +8,13 @@ package instrument
 import "github.com/dave/dst"
 
 func instrumentGin(stmt *dst.AssignStmt) []dst.Stmt {
-	if !isGin(stmt) {
-		return nil
-	}
-	stmt.Decorations().Start.Prepend(dd_instrumented)
-	return []dst.Stmt{
-		stmt,
-		ginMiddleware(stmt),
-	}
+	return instrumentMiddleware(stmt, isGin, "GinMiddleware")
 }
 
 func isGin(stmt *dst.AssignStmt) bool {
 	rhs := stmt.Rhs[0]
 	f, ok := funcIdent(rhs)
 	return ok && f.Path == "github.com/gin-gonic/gin" && (f.Name == "New" || f.Name == "Default")
-}
-
-func ginMiddleware(got *dst.AssignStmt) dst.Stmt {
-	iden, ok := got.Lhs[0].(*dst.Ident)
-	if !ok {
-		return nil
-	}
-	stmt := useMiddleware(iden.Name, "GinMiddleware")
-	markAsInstrumented(stmt)
-	return stmt
 }
 
 // removeGin returns whether a statement corresponds to orchestrion's Gin-middleware registration

--- a/internal/instrument/gin_test.go
+++ b/internal/instrument/gin_test.go
@@ -48,6 +48,7 @@ func register() {
 	}{
 		{in: `g := gin.New()`, want: `g.Use(instrument.GinMiddleware())`, tmpl: wantTpl},
 		{in: `g := gin.Default()`, want: `g.Use(instrument.GinMiddleware())`, tmpl: wantTpl},
+		{in: `api.server = gin.Default()`, want: `api.server.Use(instrument.GinMiddleware())`, tmpl: wantTpl},
 	}
 
 	for i, tc := range tests {

--- a/internal/instrument/instrument.go
+++ b/internal/instrument/instrument.go
@@ -15,6 +15,7 @@ import (
 	"log"
 	"os"
 	"path/filepath"
+	"reflect"
 	"strings"
 
 	"github.com/datadog/orchestrion/instrument/event"
@@ -571,24 +572,85 @@ func funcIdent(e dst.Expr) (*dst.Ident, bool) {
 	return f, true
 }
 
-func useMiddleware(pkg, middleware string) *dst.ExprStmt {
-	stmt := &dst.ExprStmt{
-		X: &dst.CallExpr{
-			Fun: &dst.SelectorExpr{
-				X:   &dst.Ident{Name: pkg},
-				Sel: &dst.Ident{Name: "Use"},
-			},
-			Args: []dst.Expr{
-				&dst.CallExpr{
-					Fun: &dst.Ident{
-						Name: middleware,
-						Path: "github.com/datadog/orchestrion/instrument",
+// useMiddleware returns a statement that uses the given middleware.
+func useMiddleware(expr dst.Expr, middleware string) (*dst.ExprStmt, error) {
+	/*
+		//dd:instrumented
+		r := echo.New()
+		//dd:startinstrument
+		r.Use(instrument.EchoV4Middleware())
+		//dd:endinstrument
+
+		//dd:instrumented
+		api.server = echo.New()
+		//dd:startinstrument
+		api.serverr.Use(instrument.EchoV4Middleware())
+		//dd:endinstrument
+	*/
+	var stmt *dst.ExprStmt
+	switch ex := expr.(type) {
+	case *dst.Ident:
+		stmt = &dst.ExprStmt{
+			X: &dst.CallExpr{
+				Fun: &dst.SelectorExpr{
+					X:   &dst.Ident{Name: ex.Name},
+					Sel: &dst.Ident{Name: "Use"},
+				},
+				Args: []dst.Expr{
+					&dst.CallExpr{
+						Fun: &dst.Ident{
+							Name: middleware,
+							Path: "github.com/datadog/orchestrion/instrument",
+						},
 					},
 				},
 			},
-		},
+		}
+	case *dst.SelectorExpr:
+		x, ok := ex.X.(*dst.Ident)
+		if !ok {
+			return nil, fmt.Errorf("unexpected type %v", reflect.TypeOf(ex.X))
+		}
+		stmt = &dst.ExprStmt{
+			X: &dst.CallExpr{
+				Fun: &dst.SelectorExpr{
+					X: &dst.SelectorExpr{
+						X:   &dst.Ident{Name: x.Name},
+						Sel: &dst.Ident{Name: ex.Sel.Name},
+					},
+					Sel: &dst.Ident{Name: "Use"},
+				},
+				Args: []dst.Expr{
+					&dst.CallExpr{
+						Fun: &dst.Ident{
+							Name: middleware,
+							Path: "github.com/datadog/orchestrion/instrument",
+						},
+					},
+				},
+			},
+		}
+	default:
+		return nil, fmt.Errorf("unexpected type %v", reflect.TypeOf(expr))
 	}
-	return stmt
+	markAsInstrumented(stmt)
+	return stmt, nil
+}
+
+func instrumentMiddleware(stmt *dst.AssignStmt, check func(*dst.AssignStmt) bool, middleware string) []dst.Stmt {
+	if !check(stmt) {
+		return nil
+	}
+	instrumented, err := useMiddleware(stmt.Lhs[0], middleware)
+	if err != nil {
+		fmt.Println("error instrumenting middleware", err)
+		return nil
+	}
+	stmt.Decorations().Start.Prepend(dd_instrumented)
+	return []dst.Stmt{
+		stmt,
+		instrumented,
+	}
 }
 
 func markAsWrap(stmt dst.Node) {

--- a/internal/instrument/instrument.go
+++ b/internal/instrument/instrument.go
@@ -587,51 +587,37 @@ func useMiddleware(expr dst.Expr, middleware string) (*dst.ExprStmt, error) {
 		api.server.Use(instrument.EchoV4Middleware())
 		//dd:endinstrument
 	*/
-	var stmt *dst.ExprStmt
+	var rExpr dst.Expr
 	switch ex := expr.(type) {
 	case *dst.Ident:
-		stmt = &dst.ExprStmt{
-			X: &dst.CallExpr{
-				Fun: &dst.SelectorExpr{
-					X:   &dst.Ident{Name: ex.Name},
-					Sel: &dst.Ident{Name: "Use"},
-				},
-				Args: []dst.Expr{
-					&dst.CallExpr{
-						Fun: &dst.Ident{
-							Name: middleware,
-							Path: "github.com/datadog/orchestrion/instrument",
-						},
-					},
-				},
-			},
-		}
+		rExpr = &dst.Ident{Name: ex.Name}
 	case *dst.SelectorExpr:
 		x, ok := ex.X.(*dst.Ident)
 		if !ok {
 			return nil, fmt.Errorf("unexpected type %v", reflect.TypeOf(ex.X))
 		}
-		stmt = &dst.ExprStmt{
-			X: &dst.CallExpr{
-				Fun: &dst.SelectorExpr{
-					X: &dst.SelectorExpr{
-						X:   &dst.Ident{Name: x.Name},
-						Sel: &dst.Ident{Name: ex.Sel.Name},
-					},
-					Sel: &dst.Ident{Name: "Use"},
-				},
-				Args: []dst.Expr{
-					&dst.CallExpr{
-						Fun: &dst.Ident{
-							Name: middleware,
-							Path: "github.com/datadog/orchestrion/instrument",
-						},
-					},
-				},
-			},
+		rExpr = &dst.SelectorExpr{
+			X:   &dst.Ident{Name: x.Name},
+			Sel: &dst.Ident{Name: ex.Sel.Name},
 		}
 	default:
 		return nil, fmt.Errorf("unexpected type %v", reflect.TypeOf(expr))
+	}
+	stmt := &dst.ExprStmt{
+		X: &dst.CallExpr{
+			Fun: &dst.SelectorExpr{
+				X:   rExpr,
+				Sel: &dst.Ident{Name: "Use"},
+			},
+			Args: []dst.Expr{
+				&dst.CallExpr{
+					Fun: &dst.Ident{
+						Name: middleware,
+						Path: "github.com/datadog/orchestrion/instrument",
+					},
+				},
+			},
+		},
 	}
 	markAsInstrumented(stmt)
 	return stmt, nil

--- a/internal/instrument/instrument.go
+++ b/internal/instrument/instrument.go
@@ -584,7 +584,7 @@ func useMiddleware(expr dst.Expr, middleware string) (*dst.ExprStmt, error) {
 		//dd:instrumented
 		api.server = echo.New()
 		//dd:startinstrument
-		api.serverr.Use(instrument.EchoV4Middleware())
+		api.server.Use(instrument.EchoV4Middleware())
 		//dd:endinstrument
 	*/
 	var stmt *dst.ExprStmt

--- a/samples/server/echov4.go
+++ b/samples/server/echov4.go
@@ -20,3 +20,17 @@ func echoV4Server() {
 	})
 	_ = r.Start(":8080")
 }
+
+type api struct {
+	srv *echo.Echo
+}
+
+func (a *api) echoV4Server() {
+	a.srv = echo.New()
+	a.srv.GET("/ping", func(c echo.Context) error {
+		return c.JSON(http.StatusOK, map[string]any{
+			"message": "pong",
+		})
+	})
+	_ = a.srv.Start(":8888")
+}

